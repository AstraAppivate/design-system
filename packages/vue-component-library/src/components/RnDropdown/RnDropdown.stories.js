import { storiesOf } from '@storybook/vue'

import RnDropdown from './index.vue'

storiesOf('Dropdowns', module).add('Default', () => ({
  components: { RnDropdown },
  data() {
    return {
      count: '2',
      options: [{ value: '1', label: 'One' }, { value: '2', label: 'Two' }],
    }
  },
  template: `<rn-dropdown :options="options" v-model="count" />`,
}))
