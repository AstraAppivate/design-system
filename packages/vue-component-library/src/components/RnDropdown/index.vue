<template lang="html">
  <div class="rn-dropdown" :class="open ? 'is-open' : 'is-closed'">
    <button class="button" @click="toggle">
      {{ selectedOptionLabel }}
      <svg class="h_ml-10 rn-dropdown-arrow" xmlns="http://www.w3.org/2000/svg" width="11" height="7">
        <path fill="#6F798A" fill-rule="evenodd" d="M5.66 4.49L9.19.95a1 1 0 1 1 1.42 1.41L6.36 6.61a1 1 0 0 1-1.41 0L.71 2.36A1 1 0 1 1 2.12.95l3.54 3.54z"/>
      </svg>
    </button>
    <rn-card class="rn-dropdown-menu">
      <slot v-if="!options"></slot>
      <a
        :key="index"
        v-else
        v-for="(option, index) in formattedOptions"
        @click="selectOption(option)"
      >
        {{option.label}}
      </a>
    </rn-card>
  </div>
</template>

<script src="./scripts.js"></script>
