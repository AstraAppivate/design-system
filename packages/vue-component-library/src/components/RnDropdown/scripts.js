import RnCard from '../RnCard/index.vue'

export default {
  name: 'RnDropdown',
  components: {
    RnCard,
  },

  props: {
    options: {
      type: Array,
      default: [],
    },
    value: {
      type: String,
    },
  },

  data: () => ({
    open: false,
  }),

  computed: {
    formattedOptions() {
      const { options } = this

      if (options.length > 0 && typeof options[0] === 'string') {
        return options.map(value => ({
          value,
          label: value,
        }))
      }

      return options
    },
    selectedOptionLabel() {
      const { formattedOptions, value } = this
      const selectedOption = formattedOptions.find(
        option => option.value === value
      )
      const label = selectedOption
        ? selectedOption.label
        : formattedOptions[0].label
      return label
    },
  },

  methods: {
    selectOption({ value }) {
      this.open = false
      this.$emit('input', value)
    },
    toggle() {
      this.open = !this.open
    },
  },
}
