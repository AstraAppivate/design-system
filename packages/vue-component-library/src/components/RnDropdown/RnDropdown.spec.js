import { shallow } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'

import RnDropdown from '@/src/components/RnDropdown/index.vue'

describe('RnDropdown.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(RnDropdown, {
      propsData: {
        count: '2',
        options: [{ value: '1', label: 'One' }, { value: '2', label: 'Two' }],
      },
    })
  })

  it('snapshot: has same HTML structure', () => {
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error(err)
      }

      expect(str).toMatchSnapshot()
    })
  })
})
