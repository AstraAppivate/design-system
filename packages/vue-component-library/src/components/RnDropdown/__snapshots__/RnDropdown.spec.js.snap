// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RnDropdown.vue snapshot: has same HTML structure 1`] = `
<div count="2" class="rn-dropdown is-closed"><button class="button">
    One
    <svg xmlns="http://www.w3.org/2000/svg" width="11" height="7" class="h_ml-10 rn-dropdown-arrow">
      <path fill="#6F798A" fill-rule="evenodd" d="M5.66 4.49L9.19.95a1 1 0 1 1 1.42 1.41L6.36 6.61a1 1 0 0 1-1.41 0L.71 2.36A1 1 0 1 1 2.12.95l3.54 3.54z"></path>
    </svg></button>
  <rn-card-stub class="rn-dropdown-menu"><a>
      One
    </a><a>
      Two
    </a></rn-card-stub>
</div>
`;
